core/src/main/java/io/questdb/std/ObjPriorityQueue.java

```diff
@@ -0,0 +1,154 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.std;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Comparator;
+
+public class ObjPriorityQueue<T> implements Mutable {
+    private static final int DEFAULT_CAPACITY = 16;
+    private final Comparator<T> comparator;
+    private T[] buffer;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public ObjPriorityQueue(Comparator<T> comparator) {
+        this.comparator = comparator;
+        this.buffer = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    public void add(@NotNull T value) {
+        int p = binSearch(value);
+        if (size == buffer.length) {
+            resize();
+        }
+        if (p < size) {
+            System.arraycopy(buffer, p, buffer, p + 1, size - p);
+        }
+        buffer[p] = value;
+        size++;
+    }
+
+    @Override
+    public void clear() {
+        for (int i = 0; i < size; i++) {
+            buffer[i] = null;
+        }
+        size = 0;
+    }
+
+    public T get(int index) {
+        return buffer[index];
+    }
+
+    public int getCapacity() {
+        return buffer.length;
+    }
+
+    public T poll() {
+        T v = buffer[0];
+        if (size > 0) {
+            buffer[0] = null;
+            --size;
+            System.arraycopy(buffer, 1, buffer, 0, size);
```

Comment:
Why does `pop/poll` have O(n) complexity? Should be O(ln(n)) if it followed std lib implementation. Please rename the class into something like `SortedList`.